### PR TITLE
Metric for counting  variables

### PR DESCRIPTION
**Pull request by Vitaly-Protasov**
Solved #395

### CODE REVIEW

**Comment by 0crat** [comment]
You @vitaly-protasov[/z](https://www.0crat.com/u/vitaly-protasov) need to have one of these roles in order to do "Request order start": `ARC`, `PO`; I'm sorry to say this, but at the moment you've got no roles in this project

**Comment by 0crat** [comment]
This pull request [#396](https://github.com/yegor256/aibolit/issues/396) is assigned to @lyriccoder[/z](https://www.0crat.com/u/lyriccoder), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/33aed658-0136-469b-9a5b-0083ef4e6ffa); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @acheshkov[/z](https://www.0crat.com/u/acheshkov) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by 0crat** [comment]
@0crat[/z](https://www.0crat.com/u/0crat) I see that you unassigned this issue; the order is still assigned to @lyriccoder[/z](https://www.0crat.com/u/lyriccoder) though; to cancel the order use `refuse`, as in [§6](http://www.zerocracy.com/policy.html#6)

**Comment by Vitaly-Protasov** [comment]
> We should not count the methods itself, but variables we pass in

fixed

File: aibolit/metrics/number_variables/numVariables.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from aibolit.utils.ast_builder import build_ast
+from javalang.tree import VariableDeclarator, ConstructorDeclaration
+
+
+class NumVars:
+    '''
+    This class count number of methods and variables in class
```
Comment (acheshkov) [inline aibolit/metrics/number_variables/numVariables.py:7]: wrong comment

Comment (acheshkov) [inline aibolit/metrics/number_variables/numVariables.py:7]: wrong comment: we do not calculate the number of methods

File: test/metrics/number_variables/4.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    public static Hospital getInstance() {
+        return hospital;
+    }
+
+    private Point point = new Point(800, 100);
+    private List<Bed> beds = new ArrayList<>();
+
+    private Hospital() {
+        if (Constants.BED_COUNT == 0) {
```
Comment (acheshkov) [inline test/metrics/number_variables/4.java:46]: Looks like you do not count `Constants.BED_COUNT` variable

File: aibolit/metrics/number_variables/numVariables.py (hunk 1 above)
Comment (Vitaly-Protasov) [reply aibolit/metrics/number_variables/numVariables.py:7]: yeap, sorry, forgot to rewrite it

File: test/metrics/number_variables/4.java (hunk 2 above)
Comment (Vitaly-Protasov) [reply test/metrics/number_variables/4.java:46]: fixed

**Comment by acheshkov** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 25 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: performer @lyriccoder[/z](https://www.0crat.com/u/lyriccoder)[/z](https://www.0crat.com/u/lyriccoder) didn't make CR comments

**Pull request merged**